### PR TITLE
feat(alerting): add timezone for maintenance

### LINKED ISSUE #327: Set maintenance timezone
### Describe the feature request

In addition to #312, it would be nice to specify the timezone for the window. Our maintenance windows are in local time and we would need to account for daylight saving time twice a year.

### Why do you personally want this feature to be implemented?

Ease of use ;-)

### How long have you been using this project?

2 days!

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by lefes**
## Summary
Add a "Timezone" field to the configuration in the maintenance zone to specify the timezone for the maintenance window. Also, include tests for various timezones and cases.

Fixes #327 

## Checklist
- [X] Tested and/or added tests to validate that the changes work as intended, if applicable.
- [X] Updated documentation in `README.md`, if applicable.

### CODE REVIEW

File: config/maintenance/maintenance.go [hunk 1]
```diff
@@ -85,6 +88,17 @@ func (c *Config) ValidateAndSetDefaults() error {
...
+	if c.Timezone != "" {
+		c.TimezoneLocation, err = time.LoadLocation(c.Timezone)
+		if err != nil {
+			return errInvalidTimezone
+		}
+	} else {
+		c.Timezone = "UTC"
+		c.TimezoneLocation = time.UTC
+	}
+
```
Comment (TwiN) [inline config/maintenance/maintenance.go:101]:
```suggestion
	if c.Timezone != "" {
		c.TimezoneLocation, err = time.LoadLocation(c.Timezone)
		if err != nil {
			return errInvalidTimezone
		}
	} else {
		c.Timezone = "UTC"
		c.TimezoneLocation = time.UTC
	}
```

**Review by TwiN — APPROVED** [review]

File: config/maintenance/maintenance.go [hunk 2]
```diff
@@ -12,6 +12,7 @@ var (
 	errInvalidMaintenanceStartFormat = errors.New("invalid maintenance start format: must be hh:mm, between 00:00 and 23:59 inclusively (e.g. 23:00)")
 	errInvalidMaintenanceDuration    = errors.New("invalid maintenance duration: must be bigger than 0 (e.g. 30m)")
 	errInvalidDayName                = fmt.Errorf("invalid value specified for 'on'. supported values are %s", longDayNames)
+	errInvalidTimezone               = errors.New("invalid timezone specified or format not supported. Use IANA timezone format (e.g. America/Sao_Paulo)")
```
Comment (TwiN) [inline config/maintenance/maintenance.go:15]: Can you update the README to include information about the timezone format to use?

**Review by TwiN — CHANGES_REQUESTED** [review]

File: config/maintenance/maintenance.go (hunk 2 above)
Comment (lefes) [reply config/maintenance/maintenance.go:15]: > Can you update the README to include this information about the timezone format to use?

Certainly, I've done it, but it seems a little awkward when I look at the line in the table with the parameter maintenance.timezone. I am referring to the length of the line. Is it okay, or should I change it?"

**Comment by lefes** [comment]
@TwiN Can you review his PR? I've made some changes that you are asked

File: config/maintenance/maintenance.go (hunk 2 above)
Comment (TwiN) [reply config/maintenance/maintenance.go:15]: Don't worry about it!

**Review by TwiN — APPROVED** [review]

**Comment by TwiN** [comment]
@lefes Thank you for your contribution! 🙇

**Pull request merged**